beaker/application.py

```diff
@@ -1132,17 +1132,36 @@ def build(self, client: "AlgodClient | None" = None) -> ApplicationSpecification
                 )
                 hints[abi_external.method.method_signature()] = abi_external.hints
 
-            # Compile approval and clear programs
-            approval_program, clear_program, contract = router.compile_program(
+                # Compile approval and clear programs
+            compile_results = router.compile(
+                algod_client=client,
                 version=self.build_options.avm_version,
                 assemble_constants=self.build_options.assemble_constants,
                 optimize=self.build_options.optimize_options,
+                with_sourcemaps=self.build_options.with_sourcemaps,
+                pcs_in_sourcemap=bool(client),
+                annotate_teal=self.build_options.annotate_teal,
+                annotate_teal_headers=self.build_options.annotate_teal_headers,
+                annotate_teal_concise=self.build_options.annotate_teal_concise,
             )
 
+            approval_prog: str = compile_results.approval_teal
+            clear_prog: str = compile_results.clear_teal
+
+            if compile_results.approval_sourcemap is not None:
+                approval_sm = compile_results.approval_sourcemap
+                if approval_sm.annotated_teal is not None:
+                    approval_prog = approval_sm.annotated_teal
```

Comment:
Does this mean we can compile the non-annotated one with source map and the same exact pc=>line number can be used to lookup the right line in the annotated one?

I think to use the non-annotated one I'd have to get the algokit utils repo updated to allow specifications of both versions so we can map back to the annotated one in case of LogicError

cc @daniel-makerx